### PR TITLE
Add cross building for Ubuntu ARM64

### PR DESCRIPTION
**Pull request by undef21**
Now github workflows use ubuntu 22.04 as ubuntu-latest, so we can cross-compile arm executables.

issue #4353

### CODE REVIEW

File: .github/workflows/make.yml [hunk 1]
```diff
@@ -24,14 +24,26 @@ jobs:
...
           dependencies: libsdl2-dev libsdl2-mixer-dev libsdl2-image-dev gettext
           env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, FHEROES2_WITH_IMAGE: ON, FHEROES2_WITH_DEBUG: ON, FHEROES2_WITH_TSAN: ON }
+        - name: Linux ARM64 SDL2 Release
+          os: ubuntu-latest
+          dependencies: gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libsdl2-dev:arm64 libsdl2-mixer-dev:arm64 libsdl2-image-dev:arm64 gettext libpng-dev:arm64
+          env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, CC: aarch64-linux-gnu-gcc, CXX: aarch64-linux-gnu-g++, AR: aarch64-linux-gnu-ar }
+          package_name: fheroes2_ubuntu_arm64_SDL2.zip
+          package_files: fheroes2 LICENSE script/linux/install_sdl_2.sh download_demo_version.sh extract_homm2_resources.sh changelog.txt README.txt files/lang/*.mo files/data/*.h2d
+          release_name: Ubuntu (Linux) build with SDL2 (latest commit)
+          release_tag: fheroes2-linux-sdl2_dev
```
Comment (ihhub) [inline .github/workflows/make.yml:42]: We have an old release tag used for Ubuntu ARM SDL2 builds: fheroes2-linux-arm-sdl2_dev. Please use this.

File: .github/workflows/make.yml [hunk 2]
```diff
@@ -24,14 +24,26 @@ jobs:
...
-          package_name: fheroes2_ubuntu_SDL2.zip
+          package_name: fheroes2_ubuntu_amd64_SDL2.zip
           package_files: fheroes2 LICENSE script/linux/install_sdl_2.sh download_demo_version.sh extract_homm2_resources.sh changelog.txt README.txt files/lang/*.mo files/data/*.h2d
           release_name: Ubuntu (Linux) build with SDL2 (latest commit)
           release_tag: fheroes2-linux-sdl2_dev
         - name: Linux SDL2 Debug
           os: ubuntu-latest
           dependencies: libsdl2-dev libsdl2-mixer-dev libsdl2-image-dev gettext
           env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, FHEROES2_WITH_IMAGE: ON, FHEROES2_WITH_DEBUG: ON, FHEROES2_WITH_TSAN: ON }
+        - name: Linux ARM64 SDL2 Release
```
Comment (ihhub) [inline .github/workflows/make.yml:35]: What's about SDL1?

File: .github/workflows/make.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
           os: ubuntu-latest
           dependencies: libsdl1.2-dev libsdl-mixer1.2-dev gettext
           env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, FHEROES2_WITH_SDL1: ON }
-          package_name: fheroes2_ubuntu_SDL1.zip
+          package_name: fheroes2_ubuntu_amd64_SDL1.zip
```
Comment (ihhub) [inline .github/workflows/make.yml:15]: If we update package names we should update them in the documentation files as they point to the links. However, do we need to update them? amd64 is not intuitive for every user.

File: .github/workflows/make.yml [hunk 4]
```diff
@@ -24,14 +24,26 @@ jobs:
...
           os: ubuntu-latest
           dependencies: libsdl2-dev libsdl2-mixer-dev libsdl2-image-dev gettext
           env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, FHEROES2_WITH_IMAGE: ON, FHEROES2_WITH_DEBUG: ON, FHEROES2_WITH_TSAN: ON }
+        - name: Linux ARM64 SDL2 Release
+          os: ubuntu-latest
+          dependencies: gcc-aarch64-linux-gnu g++-aarch64-linux-gnu libsdl2-dev:arm64 libsdl2-mixer-dev:arm64 libsdl2-image-dev:arm64 gettext libpng-dev:arm64
+          env: { FHEROES2_STRICT_COMPILATION: ON, FHEROES2_WITH_TOOLS: ON, CC: aarch64-linux-gnu-gcc, CXX: aarch64-linux-gnu-g++, AR: aarch64-linux-gnu-ar }
+          package_name: fheroes2_ubuntu_arm64_SDL2.zip
+          package_files: fheroes2 LICENSE script/linux/install_sdl_2.sh download_demo_version.sh extract_homm2_resources.sh changelog.txt README.txt files/lang/*.mo files/data/*.h2d
+          release_name: Ubuntu (Linux) build with SDL2 (latest commit)
```
Comment (ihhub) [inline .github/workflows/make.yml:41]: This name should have an explicit ARM word usage.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @undef21 , I put several comments here. Can you please check them?

File: .github/workflows/make.yml (hunk 1 above)
Comment (undef21) [reply .github/workflows/make.yml]: Done.

File: .github/workflows/make.yml (hunk 2 above)
Comment (undef21) [reply .github/workflows/make.yml:47]: Added SDL1 Release and Debug.

File: .github/workflows/make.yml (hunk 3 above)
Comment (undef21) [reply .github/workflows/make.yml]: Reverted package names for x86.

File: .github/workflows/make.yml (hunk 4 above)
Comment (undef21) [reply .github/workflows/make.yml]: Done.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by oleg-derevenetz** [comment]
Hi @ihhub I believe we can merge this PR since it does not affect the game code itself, just our build system.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@undef21 , thanks you very much for this change!

**Pull request merged**